addons/event/models/event_registration.py

```diff
@@ -64,46 +80,26 @@ def _check_ticket_seats_limit(self):
             if record.event_ticket_id.seats_max and record.event_ticket_id.seats_available < 0:
                 raise ValidationError(_('No more available seats for this ticket'))
 
-    @api.onchange('partner_id')
-    def _onchange_partner(self):
-        if self.partner_id:
-            self.update(self._synchronize_partner_values(self.partner_id))
+    @api.constrains('event_id', 'event_ticket_id')
+    def _check_event_ticket(self):
+        if any(registration.event_id != registration.event_ticket_id.event_id for registration in self if registration.event_ticket_id):
+            raise ValidationError(_('Invalid event / ticket choice'))
 
     # ------------------------------------------------------------
     # CRUD
     # ------------------------------------------------------------
 
     @api.model
     def create(self, vals):
```

Comment:
Could be nice to have this in create_multi mode as we could have brols that create many registrations at once.
Maybe later / if you're bored.